arch-secure-boot

```diff
@@ -75,39 +90,48 @@ case "$1" in
             --add-section .initrd=ucode-initramfs.img --change-section-vma .initrd=0x3000000 \
             /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-unsigned.efi"
 
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux="/boot/vmlinuz-$KERNEL" --change-section-vma .linux=0x40000 \
-            --add-section .initrd="/boot/initramfs-$KERNEL-fallback.img" --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-unsigned.efi"
-
-        objcopy \
-            --add-section .osrel=/etc/os-release --change-section-vma .osrel=0x20000 \
-            --add-section .linux=/boot/vmlinuz-linux-lts --change-section-vma .linux=0x40000 \
-            --add-section .initrd=/boot/initramfs-linux-lts-fallback.img --change-section-vma .initrd=0x3000000 \
-            /usr/lib/systemd/boot/efi/linuxx64.efi.stub "$NAME-recovery-lts-unsigned.efi"
+        for flavor in "${KERNEL#linux}" '-lts'; do
```

Comment:
> Allow user to choose a main kernel other than linux (for example, linux-hardened or linux-zen)

Could you please clarify a bit what you had to change? Besides converting to the `for` loop, isn't this the same as it was, you specify `KERNEL=linux-hardened` and this is the kernel that is being used? `KERNEL` parameter was precisely made to support `linux-hardened` as the default kernel, so I'm curious what was not working for you and I am maybe missing something obvious but I can't spot what functional change made it working for you...